src/ore/src/cgroup.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Linux cgroup detection utilities.
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+
+/// An entry in /proc/self/cgroup
+#[derive(Debug, PartialEq)]
+pub struct CgroupEntry {
+    subsystems: Vec<String>,
+    root: PathBuf,
+}
+
+impl CgroupEntry {
+    fn from_line(line: String) -> Option<CgroupEntry> {
+        let mut fields = line.split(':');
+        let subsystems = fields
+            .nth(1)?
+            .split(',')
+            .filter(|s| !s.is_empty())
+            .map(|s| s.to_owned())
+            .collect();
+        let root = PathBuf::from(fields.next()?);
+        Some(CgroupEntry { subsystems, root })
+    }
+}
+
+/// Parses /proc/self/cgroup into a Vec<CgroupEntry>, if the file exists.
+pub fn parse_proc_self_cgroup() -> Option<Vec<CgroupEntry>> {
+    let file = File::open("/proc/self/cgroup").ok()?;
+    let file = BufReader::new(file);
+    Some(
+        file.lines()
+            .filter_map(|result| result.ok())
```

Comment:
This should be fine as just `flatten`

```suggestion
            .flatten()
```

Reply:
Changed.